### PR TITLE
Tests for updating the display configuration while the server is running

### PR DESCRIPTION
**Pull request by mattkae**
## What's new?
- Added a single test for updating the display configuration while a display server is running
- This does not work with ubuntu frame or mir-kiosk as they don't respect the environment variable
- We can add more tests if you think this is a viable route :+1:

### CODE REVIEW

File: mir-ci/mir_ci/display_server.py [hunk 1]
```diff
@@ -60,6 +63,13 @@ def program(self, app: App, env: Dict[str, str] = {}) -> Program:
             app, env=dict({"DISPLAY": "no", "QT_QPA_PLATFORM": "wayland", "WAYLAND_DISPLAY": self.display_name}, **env)
         )
 
+    def environment(self, key: str, value: str):
+        """
+        Set an environment variable
+        """
+        self.env[key] = value
+        return self
+
```
Comment (Saviq) [inline mir-ci/mir_ci/display_server.py:72]: I would rather we add `env: Dict[str, str]` to `__init__()`, just like we have with `Program`:

https://github.com/MirServer/mir-ci/blob/a453ae04cfc644b62d785de22405596e6505736c/mir-ci/mir_ci/program.py#L43

File: mir-ci/mir_ci/test_display_configuration.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+from mir_ci.output_watcher import OutputWatcher
+
+short_wait_time = 1 * SLOWDOWN
+
+
+class TestDisplayConfiguration:
+    @pytest.mark.parametrize(
+        "local_server",
+        [
+            apps.mir_demo_server(),
```
Comment (Saviq) [inline mir-ci/mir_ci/test_display_configuration.py:18]: Leave a `TODO` please to test other servers, too.

File: mir-ci/mir_ci/test_display_configuration.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        ],
+    )
+    @pytest.mark.deps(pip_pkgs=("pyyaml",))
+    async def test_can_update_scale(self, local_server) -> None:
+        CONFIG_FILE = "/tmp/mir_ci_display_config.yaml"
+        try:
+            os.remove(CONFIG_FILE)
+        except OSError:
+            pass
```
Comment (Saviq) [inline mir-ci/mir_ci/test_display_configuration.py:27]: Best use a [tempfile.NamedTemporaryFile](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile) as a context manager (you will need to "enter" it before it gets a name).

**Review by Saviq — CHANGES_REQUESTED** [review]
Some tweaks inline

**Review by AlanGriffiths — APPROVED** [review]
We need more tests like this

**Review by Saviq — APPROVED** [review]
Not sure why you needed the decorator on each test rather than the class, but OK :)

**Pull request merged**